pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/rest/api/ComponentImpl.java

```diff
@@ -709,181 +670,157 @@ public Response deregisterFunction(final String tenant,
         try {
             requestResult = completableFuture.get();
             if (!requestResult.isSuccess()) {
-                return Response.status(Status.BAD_REQUEST).type(MediaType.APPLICATION_JSON)
-                        .entity(new ErrorData(requestResult.getMessage())).build();
+                throw new RestException(Status.BAD_REQUEST, requestResult.getMessage());
             }
         } catch (ExecutionException e) {
-            log.error("Execution Exception while deregistering {} @ /{}/{}/{}", componentType, tenant, namespace, componentName,
-                    e);
-            return Response.serverError().type(MediaType.APPLICATION_JSON)
-                    .entity(new ErrorData(e.getCause().getMessage())).build();
+            log.error("Execution Exception while deregistering {} @ /{}/{}/{}",
+                    componentType, tenant, namespace, componentName, e);
+            throw new RestException(Status.INTERNAL_SERVER_ERROR, e.getCause().getMessage());
         } catch (InterruptedException e) {
-            log.error("Interrupted Exception while deregistering {} @ /{}/{}/{}", componentType, tenant, namespace, componentName,
-                    e);
-            return Response.status(Status.REQUEST_TIMEOUT).type(MediaType.APPLICATION_JSON).build();
+            log.error("Interrupted Exception while deregistering {} @ /{}/{}/{}",
+                    componentType, tenant, namespace, componentName, e);
+            throw new RestException(Status.REQUEST_TIMEOUT, e.getMessage());
         }
-
-        return Response.status(Status.OK).entity(requestResult.toJson()).build();
     }
 
-    public Response getFunctionInfo(final String tenant,
-                                    final String namespace,
-                                    final String componentName) {
+    public FunctionConfig getFunctionInfo(final String tenant,
```

Comment:
This function should get moved out into org.apache.pulsar.functions.worker.rest.api.FunctionImpl